src/disco/containers/InstallButton.js

```diff
@@ -29,12 +32,54 @@ export class InstallButton extends React.Component {
     downloadProgress: 0,
   }
 
+  componentDidMount() {
+    const { dispatch, guid, installURL, slug } = this.props;
+    this.addonManager = new AddonManager(guid, installURL, this.statusChanged);
+    this.addonManager.getAddon()
+      .then(() => {
+        dispatch({
+          type: 'INSTALL_STATE',
+          payload: {slug, guid, url: installURL, status: INSTALLED}});
+      })
+      .catch(() => {
```

Comment:
I looked into this a bit more. The main reason to use catch instead of args is if the then case throws the `reject` callback wouldn't run. 

E.g: http://jsbin.com/luruga/edit?js,console

This raises an interesting question. If the dispatch blew-up we'd end up showing this as uninstalled. Do we want that?

One other thing is that it probably makes sense that we always return even sync functions from a then since if we forget a return and it is a promise the next then in a chain will run before that promised has resolved.

Looks like there's some eslint plugins to help with consistent promise patterns: e.g: https://github.com/xjamundx/eslint-plugin-promise which might be useful.